pkg/agent/agent.go

```diff
@@ -719,49 +719,61 @@ func (c *agentController) deleteManagedCluster(ctx context.Context, hc *hyperv1b
 		managedClusterName = hc.Name
 	}
 
-	klusterletName := "klusterlet-" + managedClusterName
-
-	// Remove the operator.open-cluster-management.io/klusterlet-hosted-cleanup finalizer in klusterlet
-	klusterlet := &operatorapiv1.Klusterlet{
+	// Delete the managed cluster
+	mc := &clusterv1.ManagedCluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: klusterletName,
+			Name: managedClusterName,
 		},
 	}
 
-	if err := c.spokeClient.Get(ctx, client.ObjectKeyFromObject(klusterlet), klusterlet); err != nil {
+	if err := c.hubClient.Get(ctx, client.ObjectKeyFromObject(mc), mc); err != nil {
 		if apierrors.IsNotFound(err) {
-			c.log.Info(fmt.Sprintf("klusterlet %v is already deleted", klusterletName))
+			c.log.Info(fmt.Sprintf("managedCluster %v is already deleted", managedClusterName))
+			mc = nil
 		} else {
-			c.log.Error(err, fmt.Sprintf("failed to get the klusterlet %v", klusterletName))
+			c.log.Info(fmt.Sprintf("failed to get the managedCluster %v", managedClusterName))
 			return err
 		}
 	}
 
-	updated := controllerutil.RemoveFinalizer(klusterlet, "operator.open-cluster-management.io/klusterlet-hosted-cleanup")
-	c.log.Info(fmt.Sprintf("klusterlet %v finalizer removed:%v", klusterletName, updated))
+	if mc != nil {
```

Comment:
Isn't `mc` always not nil because it was defined in line 723?

Reply:
I set it to nil in 732 when mc is not found. Maybe checking for err != nil would have been clearer.